tests/unit/test_install.py

```diff
@@ -0,0 +1,155 @@
+    prompts = MockPrompts(
+        {
+            r"Select the source": "0",
+            r"Do you want to Skip Validation": "yes",
+            r"Enter catalog_name": "remorph_catalog",
+            r"Enter schema_name": "remorph_schema",
+            r"Open config file in the browser.*": "yes",
+            r".*": "",
+        }
+    )
+    install = WorkspaceInstaller(prompts, mock_installation, ws)
+    webbrowser.open = lambda x: x
+    install.configure()
+
+    mock_installation.assert_file_written(
+        "config.yml",
+        {
+            "version": 1,
+            "source": "snowflake",
+            "skip_validation": True,
+            "catalog_name": "remorph_catalog",
+            "schema_name": "remorph_schema",
+        },
+    )
+
+
+def test_create_catalog_schema(ws, mock_installation):
+    prompts = MockPrompts(
+        {
+            r"Select the source": "0",
+            r"Do you want to Skip Validation": "yes",
+            r"Enter catalog_name": "test",
+            r".*Do you want to create a new one?": True,
+            r"Enter schema_name": "schema",
+            r".*Do you want to create a new Schema?": True,
+            r".*": "",
+        }
+    )
+    install = WorkspaceInstaller(prompts, mock_installation, ws)
+    install._catalog_setup._ws.catalogs.get.side_effect = NotFound("test")
+    install._catalog_setup._ws.schemas.get.side_effect = NotFound("schema")
+    install.configure()
+
+
+def test_create_catalog_no(ws, mock_installation):
+    prompts = MockPrompts(
+        {
+            r"Select the source": "0",
+            r"Do you want to Skip Validation": "yes",
+            r"Enter catalog_name": "test",
+            r".*Do you want to create a new one?": False,
+            r".*": "",
+        }
+    )
+    install = WorkspaceInstaller(prompts, mock_installation, ws)
+    install._catalog_setup._ws.catalogs.get.side_effect = NotFound("test")
+    with pytest.raises(SystemExit):
+        install.configure()
+
+
+def test_create_schema_no(ws, mock_installation):
+    prompts = MockPrompts(
+        {
+            r"Select the source": "0",
+            r"Do you want to Skip Validation": "yes",
+            r"Enter catalog_name": "test",
+            r".*Do you want to create a new one?": True,
+            r"Enter schema_name": "schema",
+            r".*Do you want to create a new Schema?": False,
+            r".*": "",
+        }
+    )
+    install = WorkspaceInstaller(prompts, mock_installation, ws)
+    install._catalog_setup._ws.schemas.get.side_effect = NotFound("test.schema")
+    with pytest.raises(SystemExit):
+        install.configure()
+
+
+def test_workspace_installation(ws, mock_installation, monkeypatch):
+    # Create a mock for the current method of the Installation
+    mock_current = MagicMock(return_value=mock_installation)
+
+    # Set the return value of the current method of the Installation mock
+    mock_installation.current = mock_current
```

Comment:
Current is a static method - you can't mock that

Reply:
Updated the code